src/NuGet.Services.EndToEnd/Support/Clients/V2V3SearchClient.cs

```diff
@@ -234,36 +234,68 @@ public async Task<IReadOnlyList<SearchServiceProperties>> GetSearchServicesAsync
 
         private async Task<SearchServiceProperties> GetSearchServiceFromAzureAsync(ServiceDetails serviceDetails, ITestOutputHelper logger)
         {
-            if (serviceDetails.UseConfiguredUrls)
+            // If the configs contain only a hardcoded URL, return the hardcoded URL.
+            // If the configs contain only cloud service details, look up URL and instance count from Azure.
+            // If the configs contain both a hardcoded URL and cloud service details, look up instance count from Azure
+            // but override the URL with the hardcoded URL.
+            string hardcodedUrl;
+            if (serviceDetails.SearchBaseUrl != null)
```

Comment:
There will be a follow-up PR to remove staging URL and production URL (in lieu of `SearchBaseUrl`) and clean up `UseOfficialDns` too. The config is very confusing right now and needs some work...